### PR TITLE
Rar support

### PR DESCRIPTION
**Pull request by limelime**

### CODE REVIEW

**Comment by delcypher** [comment]
@limelime This looks like a very valuable contribution. Thanks! I've been meaning to write tests but I never really got around to it.

I want to test this a bit first but once I'm happy I'll go ahead and merge unless @arfoll has any objections.

File: unrarall [hunk 1]
```diff
@@ -149,6 +149,9 @@ function getUnrarFlags()
     unrar)
       echo -o+ -y
     ;;
+    rar)
+      echo -o+ -y
+    ;;    
     7z)
```
Comment (arfoll) [inline unrarall]: Trailing whitespace here and a few other spots

**Comment by arfoll** [comment]
Generally this looks great. If you want to give it a qiuck spin before a merge then awesome. I love the test scripts, could have a really cool travis job :). There's a few things before I think it should get merged:

1. The whitespace and the fact fce58e2 should get rebased into ffa584
2. I don't like a2865c1 it completely breaks history and well a vimtype is probably much smarter to get highlight?
3. ffa5849 should be rebased with 386cba3 since they do the same thing?

Essentially a bit of git cleanup required me thinks! I'm happy to do it, the testing the is the big bit really.

I'd played before with using rar instead of unrar, but didn't see any performance benefits and with the nonfree status didn't see the point compared to 7z. But I'm curious @limelime what's your reason for adding rar support?

**Comment by delcypher** [comment]
@arfoll @limelime I've just landed a bunch of changes to do much better testing. I opted to use `lit` rather than write tests directly in bash so that I can

* Run tests in parallel
* Selectively enable/disable tests depending on the testing environment. This is very useful if certain features don't work with a particular backend.

Your existing tests were a very handy guide on how to use `rar` so don't feel bad that I didn't use your tests.

Now that we have better testing this PR needs to be reworked to work with the new tests. Specifically to fit in with testing.

1. Modify `tests/lit.site.cfg` to be aware of the `rar` backend and fail if the `rar` binary is not available if that backend is requested. You will need to add a flag to `unrarall` similar to `--7zip` which forces the `rar` backend to be used.
2. Modify `tests/run.sh` to run `lit -v --param unrarall_backend=rar`.

So I guess the first step is to rework the core change (supporting `rar` as a backend) into its own commit and then add testing support.

**Comment by delcypher** [comment]
@arfoll @limelime Seeing as unrarall's code is fresh in my mind I thought I'd just implement `rar` support. I've done this in 3ea206b2e7ad7a5f9b83806f07f143e1cc5d0e01 .

@limelime Thank you for your work here. I'm closing this issue but if you find additional problems please file and issue or create a PR :)

**Pull request closed without merging**